handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -614,21 +614,34 @@ public final SSLEngineResult wrap(
                 }
             }
 
-            if (dst.remaining() < MAX_ENCRYPTED_PACKET_LENGTH) {
+            int endOffset = offset + length;
+            int srcsLen = 0;
+
+            for (int i = offset; i < endOffset; ++i) {
+                final ByteBuffer src = srcs[i];
+                if (src == null) {
+                    throw new IllegalArgumentException("srcs[" + i + "] is null");
+                }
+                srcsLen += src.remaining();
+            }
+
+            int maxEncryptedLen = calculateOutNetBufSize(srcsLen);
+
+            // If overflow happened just use MAX_PLAINTEXT_LENGTH.
+            if (maxEncryptedLen < 0 || MAX_ENCRYPTED_PACKET_LENGTH < maxEncryptedLen) {
```

Comment:
you mean the overflow ? We may not correctly return BUFFER_OVERFLOW. This will only happen tho if the sum of the remaining bytes in srcs is super big. This should not happen in practice.